### PR TITLE
[DataTable] Allow overriding row styles from app

### PR DESCRIPTION
**Pull request by SferaDev**
New API: ``rowConfig?(row: T): RowConfig;``

RowConfig:

```
export interface RowConfig {
    style?: CSSProperties;
    disabled?: boolean;
    selectable?: boolean;
}
```

Usage example:

```
rowConfig={(_row: T) => ({
    style: { backgroundColor: "yellow"},
    disabled: true
})}
```

Behavior changes:

- Selectable: Disables the checkbox
- Disabled: Greys-out and disables pointer events (if selectable not set also disables the checkbox)
- Style: Overrides all pre-set styling in row (including changes applied by disabled/selectable)

### CODE REVIEW

**Comment by adrianq** [comment]
I guess this is ready to be merged, right @SferaDev? It seems to be used at https://github.com/EyeSeeTea/metadata-synchronization/pull/489

File: src/data-table/DataTableBody.tsx [hunk 1]
```diff
@@ -38,17 +46,21 @@ const useStyles = makeStyles({
         background: "#E0E0E0",
     },
     disabledRow: {
-        backgroundColor: "#F5DFDF !important",
+        backgroundColor: "#EEEEEE",
```
Comment (adrianq) [inline src/data-table/DataTableBody.tsx:49]: out of curiosity, why this change in the background color for disabled rows?

**Comment by adrianq** [comment]
@tokland This seems to be working fine for metadata sync. Just wondering, if it has any impact on project monitoring. Can you have a look at it? If you haven't done it yet....

File: src/data-table/DataTableBody.tsx (hunk 1 above)
Comment (tokland) [reply src/data-table/DataTableBody.tsx:49]: @SferaDev  I guess now we should set the background color from the app `rowConfig`, now that we can? Although, maybe we could leave some (sane) different color, as default?

**Comment by SferaDev** [comment]
I noticed that **Select all** includes disabled rows, I will add a fix for that to exclude them

File: src/data-table/DataTableBody.tsx (hunk 1 above)
Comment (SferaDev) [reply src/data-table/DataTableBody.tsx:49]: Exactly, I set ``grey`` color for disabled as it's the default in material and in the app we can now override the styling as we want.

File: src/data-table/DataTableBody.tsx [hunk 2]
```diff
@@ -158,6 +171,7 @@ export function DataTableBody<T extends ReferenceObject>(props: DataTableBodyPro
                     onContextMenu={event => handleClick(event)}
                     role="checkbox"
                     selected={checked || indeterminate}
+                    style={style}
```
Comment (tokland) [inline src/data-table/DataTableBody.tsx:172]: I am unable to make this work, I pass a style from the app, and it's indeed in this `style`, but it's not present on the `<tr>`. Am I missing something?

Comment (SferaDev) [reply src/data-table/DataTableBody.tsx:172]: Not sure, I am able to do it on ``metadata-sync`` and I did not require the ``!important``, Perhaps ``project-monitoring`` has something in the global stylesheet?

Comment (tokland) [reply src/data-table/DataTableBody.tsx:172]: It was me, I had to remove the "!important", cannot be used in an inline css. Working fine!

**Review by tokland — APPROVED** [review]
Code +:1+

https://github.com/EyeSeeTea/project-monitoring-app/pull/217
